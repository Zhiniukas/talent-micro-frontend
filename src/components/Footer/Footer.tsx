import type { FC } from "react";

export const Footer: FC = () => {
  const currentYear = new Date().getFullYear();

  return (
    <footer>
      <p aria-label="current year">{currentYear}</p>
    </footer>
  );
};
