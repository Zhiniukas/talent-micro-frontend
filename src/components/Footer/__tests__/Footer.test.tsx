import { render, screen } from "@testing-library/react";
import { Footer } from "../Footer";
import renderer from "react-test-renderer";

describe("Footer", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<Footer />);

    expect(tree).toMatchSnapshot();
  });

  it("should render current year", () => {
    render(<Footer />);
    
    const currentYearElement = screen.getByLabelText("current year");

    expect(currentYearElement).toBeVisible();
    expect(currentYearElement.textContent).toBe(`${new Date().getFullYear()}`);
  });
});
