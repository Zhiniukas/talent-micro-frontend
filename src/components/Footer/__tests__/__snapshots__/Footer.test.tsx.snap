// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<footer>
  <p
    aria-label="current year"
  >
    2023
  </p>
</footer>
`;
